airflow/decorators/task_group.py

```diff
@@ -21,28 +21,60 @@
 """
 import functools
 from inspect import signature
-from typing import Callable, Optional, TypeVar, cast
+from typing import TYPE_CHECKING, Any, Callable, Dict, Optional, TypeVar, overload
 
 from airflow.utils.task_group import TaskGroup
 
-T = TypeVar("T", bound=Callable)
+if TYPE_CHECKING:
+    from airflow.models import DAG
+
+F = TypeVar("F", bound=Callable[..., Any])
 
 task_group_sig = signature(TaskGroup.__init__)
 
 
-def task_group(python_callable: Optional[Callable] = None, *tg_args, **tg_kwargs) -> Callable[[T], T]:
+# This covers the @task_group() case. Annotations are copied from the TaskGroup
```

Comment:
Nice. I was sure I saw the recipe somewhere :) https://mypy.readthedocs.io/en/stable/generics.html#declaring-decorators

Thanks!. The Type Hiting of Python is an interesting one.